### PR TITLE
Include full stacktraces from LogTestDurationListener test hang monitor

### LINKED ISSUE #6202: Include full stacktraces from LogTestDurationListener test hang monitor
See https://github.com/prestosql/presto/runs/1497472672

- tests apparently hanged
- one test thread is here:

```
2020-12-04T07:53:43.1535741Z "pool-2-thread-2" prio=5 Id=16 BLOCKED on org.testng.TestClass@2a78dd6a owned by "pool-2-thread-1" Id=15
2020-12-04T07:53:43.1537790Z 	at app//org.testng.internal.TestMethodWorker.invokeBeforeClassMethods(TestMethodWorker.java:158)
2020-12-04T07:53:43.1539747Z 	-  blocked on org.testng.TestClass@2a78dd6a
2020-12-04T07:53:43.1540972Z 	at app//org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
2020-12-04T07:53:43.1542880Z 	at java.base@11.0.9/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
2020-12-04T07:53:43.1545474Z 	at java.base@11.0.9/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
2020-12-04T07:53:43.1547540Z 	at java.base@11.0.9/java.lang.Thread.run(Thread.java:834)
2020-12-04T07:53:43.1548037Z 
2020-12-04T07:53:43.1548486Z 	Number of locked synchronizers = 1
2020-12-04T07:53:43.1549747Z 	- java.util.concurrent.ThreadPoolExecutor$Worker@7cedfa63
```

- the information who owns `TestClass@2a78dd6a` is not present, probably because every stacktrace is limited to 8 frames.

see full thread dump -- https://github.com/prestosql/presto/runs/1497472672  or  https://gist.github.com/findepi/2f87bad4047976bee6af9223f5d3ff6c

### PR DESCRIPTION
**Pull request by aj-rom**
Fixes #6202

### CODE REVIEW

File: presto-testng-services/src/main/java/io/prestosql/testng/services/StackTraceUtil.java [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+        return sb;
+    }
+
+    /**
+     * Returns the full stacktrace string of the thread
+     * @param threadInfo the thread information
+     * @return the full stacktrace of the thread information
+     */
+    public static String getFullStackTrace(ThreadInfo threadInfo)
```
Comment (kokosing) [inline presto-testng-services/src/main/java/io/prestosql/testng/services/StackTraceUtil.java:139]: Please put public at the top, let's order method by usage. So first public, then methods used in public method, then methods that were used by methods above...

File: presto-testng-services/src/main/java/io/prestosql/testng/services/StackTraceUtil.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+ */
+
+package io.prestosql.testng.services;
+
+import java.lang.management.LockInfo;
+import java.lang.management.MonitorInfo;
+import java.lang.management.ThreadInfo;
+import java.time.Duration;
+
+public final class StackTraceUtil
```
Comment (kokosing) [inline presto-testng-services/src/main/java/io/prestosql/testng/services/StackTraceUtil.java:22]: Can you please add tests, so:
 - we are sure it works in the expected way
 - we see how new stack trace information is going to look like.


Is the format you are producing acceptable by:
 - https://spotify.github.io/threaddump-analyzer/
 - https://fastthread.io/
?

File: presto-testng-services/src/main/java/io/prestosql/testng/services/StackTraceUtil.java [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+{
+    private StackTraceUtil() {}
+
+    /**
+     * The header section of the thread
+     * @param threadInfo the thread information
+     * @return the string builder of the header starting at a new line.
+     */
```
Comment (kokosing) [inline presto-testng-services/src/main/java/io/prestosql/testng/services/StackTraceUtil.java:30]: By the rule of thumb we avoid comments. We try to comment that are part of SPI or things that do complex and unexpected things.

File: presto-testng-services/src/main/java/io/prestosql/testng/services/LogTestDurationListener.java [hunk 4]
```diff
@@ -145,8 +144,7 @@ private static void dumpAllThreads(String message)
     {
         LOG.warn("%s\n\nFull Thread Dump:\n%s", message,
                 Arrays.stream(getThreadMXBean().dumpAllThreads(true, true))
-                        // TODO ThreadInfo.toString truncates stacktrace to java.lang.management.ThreadInfo#MAX_FRAMES
-                        .map(ThreadInfo::toString)
```
Comment (findepi) [inline presto-testng-services/src/main/java/io/prestosql/testng/services/LogTestDurationListener.java:149]: There is a similar code in `io.prestosql.tests.product.launcher.cli.SuiteRun.Execution#reportSuspectedTimeout`. 
Can you please update it too?

File: presto-testng-services/src/main/java/io/prestosql/testng/services/StackTraceUtil.java (hunk 2 above)
Comment (findepi) [reply presto-testng-services/src/main/java/io/prestosql/testng/services/StackTraceUtil.java:22]: a test would also help review (this & any future changes), as you can then see how the output is changing

see eg https://github.com/airlift/stackfold/blob/master/src/test/java/io/airlift/stackfold/TestStackFolder.java

File: presto-testng-services/src/main/java/io/prestosql/testng/services/LogTestDurationListener.java (hunk 4 above)
Comment (CoachLuck) [reply presto-testng-services/src/main/java/io/prestosql/testng/services/LogTestDurationListener.java:149]: For sure and thank you for pointing me to the right location :), I apologize for slow responses as I have been pretty busy with finals this week, so I will hopefully be getting around to this come next week :)

File: presto-testng-services/src/main/java/io/prestosql/testng/services/StackTraceUtil.java (hunk 1 above)
Comment (CoachLuck) [reply presto-testng-services/src/main/java/io/prestosql/testng/services/StackTraceUtil.java:139]: Gotchya,  I can push these changes soon.

**Comment by CoachLuck** [comment]
@findepi @kokosing thanks for taking the time to review this.

Finally getting free time late tomorrow night and will have access to a Mac to be able to run tests, etc.

Apologize for the delayed response, finals week got slightly stressful 😁

**Comment by bitsondatadev** [comment]
:wave: @aj-rom - this PR is inactive and doesn't seem to be under development, and it might already be implemented by #8179. If you'd like to continue work on this at any point in the future, feel free to re-open.

**Pull request closed without merging**